samples/Nancy.Swagger.Annotations.Demo/Modules/GenericModule.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Collections.Generic;
+using Microsoft.Extensions.FileProviders;
+using Nancy.ModelBinding;
+using Nancy.Responses.Negotiation;
+using Nancy.Swagger.Demo.Models;
+using Nancy.Swagger.Annotations.Attributes;
+using Swagger.ObjectModel;
+using Nancy.Swagger.Services;
+
+namespace Nancy.Swagger.Demo.Modules
+{
+    public class GenericModule : NancyModule
+    {
+        public GenericModule(ISwaggerTagCatalog tagCatalog)
+            : base("/generic")
+        {
+            tagCatalog.AddTag(new Tag()
+            {
+                Description = "All response are wrapped in a base class that can include metadata",
+                Name = "Generics"
+            });
+
+            Get("/users", _ => GetUsers(), null, "GetUsers");
+
+            Post("/users", _ =>
+            {
+                var user = this.Bind<User>();
+                return PostUser(user);
+            });
+
+            Get("/multi-generics", _ => GetMultipleGenerics(), null, "GetMultipleGenerics");
+            Get("/multi-nested-generics", _ => GetMultipleNestedGenerics(), null, "GetMultipleNestedGenerics");
+        }
+
+        [Route("GetMultipleGenerics")]
+        [Route(Summary = "Get a generic response with multiple types")]
+        [SwaggerResponse(HttpStatusCode.OK, Message = "OK", Model = typeof(MultipleGenericResponse<string, int, double>))]
+        [Route(Tags = new[] { "Generics" })]
+        private MultipleGenericResponse<string, int, double> GetMultipleGenerics()
+        {
+            return new MultipleGenericResponse<string, int, double>("A string!", 1, 42.5);
+        }
+
+        [Route("GetMultipleNestedGenerics")]
+        [Route(Summary = "Get crazy nested generic response")]
+        [SwaggerResponse(HttpStatusCode.OK, Message = "OK", Model = typeof(MultipleGenericResponse<ApiResponse<ApiResponse<string>>, MultipleGenericResponse<string, int, double>, long>))]
+        [Route(Tags = new[] { "Generics" })]
+        private MultipleGenericResponse<ApiResponse<ApiResponse<string>>, MultipleGenericResponse<string, int, double>, long> GetMultipleNestedGenerics()
```

Comment:
Just commenting because I thought it was funny:
`MultipleGenericResponseOfApiResponseOfApiResponseOfStringAndMultipleGenericResponseOfStringAndInt32AndDoubleAndInt64`